core/v1/extension_types.go

```diff
@@ -107,6 +107,13 @@ func (smec *ServiceMeshExtensionConfig) DeepCopy() *ServiceMeshExtensionConfig {
 }
 
 func (smec *ServiceMeshExtensionConfig) UnmarshalJSON(in []byte) error {
+	if len(in) == 0 {
+		if smec.Data != nil {
```

Comment:
Do we always want to set `Data` to a new map?

Reply:
hmm, good question. I can't remember why I had put this. Removed, tests pass. Thanks.